File: java/src/main/java/cn/huidu/device/api/demo/demo2_task/Scheduled.java
package cn.huidu.device.api.demo.demo2_task;

import java.util.HashMap;
import java.util.Map;

import cn.huidu.device.api.sdk.data.task.ScheduledTaskInfo;
import cn.huidu.device.api.sdk.deviceTask.ScheduledTask;

public class Scheduled {
    public static void main(String[] args) {
        // 替换成你自己的参数
        // String host = "${YourHost}";
        // String sdkKey = "${YourSdkKey}";
        // String sdkSecret = "${YourSdkSecret}";
        // Config.InitSdk(host, sdkKey, sdkSecret);

        // 1. 实例化任务接口对象 
        ScheduledTask scheduledTask = new ScheduledTask();

        Map<String, ScheduledTaskInfo[]> taskInfos = new HashMap<>();

        ScheduledTaskInfo taskInfo = new ScheduledTaskInfo();
        taskInfo.setTimeRange("06:00:00~19:00:00");
        taskInfos.put("screen", new ScheduledTaskInfo[] {taskInfo});
        // 2. 调用接口
        
        // 切换南区域为1
        String jsonResult = scheduledTask.setScheduledTask("", taskInfos);
        System.out.println(jsonResult);
    }

}

File: python/demos/demo2_task/Scheduled.py
from sdk.common.Config import Config
from sdk.deviceTask.ScheduledTask import ScheduledTask
from sdk.data.task.ScheduledTaskInfo import ScheduledTaskInfo
import json

def main():
    # with your own parameters
    # host = "${YourHost}"
    # sdk_key = "${YourSdkKey}"
    # sdk_secret = "${YourSdkSecret}"
    # Config.init_sdk(host, sdk_key, sdk_secret)

    # Instantiate task interface objects
    scheduled_task = ScheduledTask()

    # Prepare the task configuration
    task_info = ScheduledTaskInfo()
    task_info.time_range = "06:00:00~19:00:01"  # Set the time frame

    task_infos = {
        "screen": [task_info]  # The task force is called "screen"
    }

    result = scheduled_task.set_scheduled_task("C16L-D00-A000F", task_infos)
    print("Scheduled task setting results:")
    print(json.dumps(result, indent=2))

if __name__ == "__main__":
    main()
